enumarable_methods.rb

```diff
@@ -0,0 +1,74 @@
+module Enumerable
+  def my_each
+    each do |i|
+      yield i
+      return to_enum unless block_given?
+    end
+  end
+
+  def my_each_with_index
+    (0...length).each do |i|
+      yield(self[i], i)
+    end
+  end
+
+  def my_select
+    selection = []
+    my_each { |val| selection.push(val) if yield(val) }
+    selection
+  end
+end
```

Comment:
You closed your enumerable module with this end. You should delete it and move it to the end.